ee/libcglue/src/glue.c

```diff
@@ -134,132 +130,17 @@ void compile_time_check() {
 }
 #endif
 
-#ifdef F___normalized_path
-/* Sanitize a pathname by removing . and .. components, duplicated /, etc. */
-static char* sanitize_path(const char *path_name)
-{
-	int i, j;
-	int first, next;
-	static char out[255];
-
-	/* First copy the path into our temp buffer */
-	strcpy(out, path_name);
-	/* Then append "/" to make the rest easier */
-	strcat(out,"/");
-
-	/* Convert "//" to "/" */
-	for(i=0; out[i+1]; i++) {
-		if(out[i]=='/' && out[i+1]=='/') {
-			for(j=i+1; out[j]; j++)
-					out[j] = out[j+1];
-			i--;
-		;}
-	}
-
-	/* Convert "/./" to "/" */
-	for(i=0; out[i] && out[i+1] && out[i+2]; i++) {
-		if(out[i]=='/' && out[i+1]=='.' && out[i+2]=='/') {
-			for(j=i+1; out[j]; j++)
-					out[j] = out[j+2];
-			i--;
-		}
-	}
-
-	/* Convert "/path/../" to "/" until we can't anymore.  Also convert leading
-	 * "/../" to "/" */
-	first = next = 0;
-	while(1) {
-		/* If a "../" follows, remove it and the parent */
-		if(out[next+1] && out[next+1]=='.' &&
-			out[next+2] && out[next+2]=='.' &&
-			out[next+3] && out[next+3]=='/') {
-			for(j=0; out[first+j+1]; j++)
-				out[first+j+1] = out[next+j+4];
-			first = next = 0;
-			continue;
-		}
-
-		/* Find next slash */
-		first = next;
-		for(next=first+1; out[next] && out[next] != '/'; next++)
-			continue;
-		if(!out[next]) break;
-	}
-
-	/* Remove trailing "/" */
-	for(i=1; out[i]; i++)
-		continue;
-	if(i >= 1 && out[i-1] == '/')
-		out[i-1] = 0;
-
-	return (char*)out;
-}
-
-static int isCdromPath(const char *path)
-{
-	return !strncmp(path, "cdrom0:", 7) || !strncmp(path, "cdrom:", 6);
-}
-
-char *__normalized_path(const char *originalPath)
-{
-	const char *buf = sanitize_path(originalPath);
-	static char b_fname[FILENAME_MAX];
-
-	if (!strchr(buf, ':')) { // filename doesn't contain device
-		if (buf[0] == '/' || buf[0] == '\\') {   // does it contain root ?
-			char *device_end = strchr(__direct_pwd, ':');
-			if (device_end) {      // yes, let's strip pwd a bit to keep device only
-				strncpy(b_fname, __direct_pwd, device_end - __direct_pwd);
-				strcpy(b_fname + (device_end - __direct_pwd), buf);
-			} else {               // but pwd doesn't contain any device, let's default to host
-				strcpy(b_fname, "host:");
-				strcpy(b_fname + 5, buf);
-			}
-		} else {                 // otherwise, it's relative directory, let's copy pwd straight
-			int b_fname_len = strlen(__direct_pwd);
-			if (!strchr(__direct_pwd, ':')) { // check if pwd contains device name
-				strcpy(b_fname, "host:");
-				strcpy(b_fname + 5, __direct_pwd);
-				if (!(__direct_pwd[b_fname_len - 1] == '/' || __direct_pwd[b_fname_len - 1] == '\\')) { // does it has trailing slash ?
-					if(isCdromPath(b_fname)) {
```

Comment:
Is this thing really needed?
I don't know why but it compares If unit is `cdrom` and then adding `\\` is that something legacy? maybe from windows?

@uyjulian @rickgaiser , do you know something about this?

Reply:
This is the correct behavior. In the versions of cdvdman I've checked `CdSearchFile` checks for backslashes (and not forward slashes)